optuna/storages/_rdb/storage.py

```diff
@@ -773,27 +774,12 @@ def _check_and_set_param_distribution(
 
             models.TrialModel.find_or_raise_by_id(trial_id, session)
 
-            previous_record = (
-                session.query(models.TrialParamModel)
-                .join(models.TrialModel)
-                .filter(models.TrialModel.study_id == study_id)
-                .filter(models.TrialParamModel.param_name == param_name)
-                .first()
-            )
-            if previous_record is not None:
-                distributions.check_distribution_compatibility(
-                    distributions.json_to_distribution(previous_record.distribution_json),
-                    distribution,
-                )
-
-            session.add(
-                models.TrialParamModel(
-                    trial_id=trial_id,
-                    param_name=param_name,
-                    param_value=param_value_internal,
-                    distribution_json=distributions.distribution_to_json(distribution),
-                )
-            )
+            models.TrialParamModel(
+                trial_id=trial_id,
+                param_name=param_name,
+                param_value=param_value_internal,
+                distribution_json=distributions.distribution_to_json(distribution),
+            ).check_and_add(session)
```

Comment:
Not introduced by this PR but I'm wondering whether this doesn't cause a violation of the uniqueness constraint on insertion.